addons/binding/org.openhab.binding.fronius/src/main/java/org/openhab/binding/fronius/handler/FroniusHandler.java

```diff
@@ -0,0 +1,369 @@
+            case FroniusBindingConstants.InverterDataChannelUac:
+                return inverterRealtimeResponse.getBody().getData().getUac();
+            case FroniusBindingConstants.InverterDataChannelUdc:
+                return inverterRealtimeResponse.getBody().getData().getUdc();
+        }
+
+        if (powerFlowResponse == null) {
+            return null;
+        }
+        switch (fieldName) {
+            case FroniusBindingConstants.PowerFlowpGrid:
+                return powerFlowResponse.getBody().getData().getSite().getPgrid();
+            case FroniusBindingConstants.PowerFlowpLoad:
+                return powerFlowResponse.getBody().getData().getSite().getPload();
+            case FroniusBindingConstants.PowerFlowpAkku:
+                return powerFlowResponse.getBody().getData().getSite().getPakku();
+        }
+
+        return null;
+    }
+
+    /**
+     * Get new data
+     *
+     */
+    private void updateData() {
+        FroniusConfiguration config = getConfigAs(FroniusConfiguration.class);
+        inverterRealtimeResponse = getRealtimeData(config.ip, config.deviceId);
+        powerFlowResponse = getPowerFlowRealtime(config.ip);
+    }
+
+    /**
+     * Make the PowerFlowRealtimeDataRequest
+     *
+     * @param ip
+     * @return {PowerFlowRealtimeResponse}
+     */
+    private PowerFlowRealtimeResponse getPowerFlowRealtime(String ip) {
+        PowerFlowRealtimeResponse result = null;
+        boolean resultOk = false;
+        String errorMsg = null;
+
+        try {
+
+            String location = PowerFlowRealtimeData.replace("%IP%", StringUtils.trimToEmpty(ip));
+            logger.debug("URL = {}", location);
+            URL url = new URL(location);
+            URLConnection connection = url.openConnection();
+
+            try {
+                String response = IOUtils.toString(connection.getInputStream());
+                logger.debug("aqiResponse = {}", response);
+                result = gson.fromJson(response, PowerFlowRealtimeResponse.class);
+            } finally {
+                IOUtils.closeQuietly(connection.getInputStream());
+            }
+
+            if (result == null) {
+                errorMsg = "no data returned";
+            } else if (result.getBody() != null) {
+                resultOk = true;
+            } else {
+                errorMsg = "missing data sub-object";
+            }
+
+            if (!resultOk) {
+                logger.warn("Error in fronius response: {}", errorMsg);
+            }
+
+        } catch (MalformedURLException e) {
+            errorMsg = e.getMessage();
+            logger.warn("Constructed url is not valid: {}", errorMsg);
+        } catch (JsonSyntaxException e) {
+            errorMsg = "Configuration is incorrect";
+            logger.warn("Error running fronius request: {}", errorMsg);
+        } catch (IOException | IllegalStateException e) {
+            errorMsg = e.getMessage();
```

Comment:
What are you doing with `errorMsg` after you have filled it? don't you want to log it? :)

Reply:
@triller-telekom the fronius device goes offline when the sun does not shine. The device isn't available at this time. Do you have any idea how i could managed this in openhab? 
If i can't connect to the device/api it does not have to be a mistake. 
that's the reason for the missing log at this point.

Reply:
@gbeine I have no storage and at night my device completly shutdown. The ip isn't available at night. The inverter have connection to the power network and is still avaible for some time if the sun does not shin but after a few minutes it switch off.

Reply:
Don't want to interfere, I agree with trokohl - my inverter is also offline when no power is produced. I could imagine, that the API is available when the "Nachtmodus" is turned on. This mode is mentioned in the Symo user manual, but I never tried enabling it because it doesn't make sense to me "wasting" energy just for getting some data ;-) although the data might of course differ a little bit from the "real" values because the inverter got offline since the last request.
But regarding the errorMsg: what about just logging it on trace level?

Reply:
you're right i can log on trace level, but it's an error if it is the wrong ip. maybe i log the first exception as an error and all following equal exceptions on trace level.

Reply:
Sounds good I think - maybe addditionally (or an alternative way): log it as an error only if the request did never succeed before (and the IP was "validated" that way).
In a future version one could then also add an "unreachable" channel which e.g.  is set to "true" if the request did never succeed within the last 24h or so.